charger/ocpp/connector.go

```diff
@@ -128,40 +128,24 @@ func (conn *Connector) TransactionID() (int, error) {
 	return conn.txnId, nil
 }
 
-func (conn *Connector) Status() (api.ChargeStatus, error) {
+// Status returns the unmapped charge point status
+func (conn *Connector) Status() (core.ChargePointStatus, error) {
+	if !conn.cp.Connected() {
+		return "", api.ErrTimeout
+	}
+
 	conn.mu.Lock()
 	defer conn.mu.Unlock()
 
-	res := api.StatusNone
-
-	if !conn.cp.Connected() {
-		return res, api.ErrTimeout
+	if conn.status == nil {
+		return core.ChargePointStatusUnavailable, nil
 	}
 
```

Comment:
Das kann man so nicht machen, denn der Status des Ladepunkts _heisst_ `Unavailable`. Das ist trotzdem ein valider Status den man u.a. auch gezielt setzen kann (z. B. zu Wartungsarbeiten) damit keine neuen Ladevorgänge begonnen werden oder der Nutzer zumindest eine Information erhalten kann. Wenn ein CP die Verbindung verliert wechselt er _auch_ automatisch in diesen Status.

Dies ist jedoch kein Fehlerzustand wie "unbekannter Zustand".
```suggestion
```

Reply:
Das mag sein. Aber kein Status gibt ein panic. Das muss vorher abfangen werden, wie auch immer das umgesetzt wird.